services/datamanager/builtin/builtin.go

```diff
@@ -422,6 +455,16 @@ func (svc *builtIn) Reconfigure(
 		fileLastModifiedMillis = defaultFileLastModifiedMillis
 	}
 
+	svc.selectiveSync = svcConfig.SelectiveSync
+	if svc.selectiveSync && svcConfig.SelectiveSyncer != "" {
+		syncSensor, err := sensor.FromDependencies(deps, svcConfig.SelectiveSyncer)
```

Comment:
So I understand how we grab the correct selective resource, in https://github.com/viam-labs/trigger-sync-examples-v2/blob/main/sync_module/sync-on-color.json, how is `selective-syncer` tying into `test-module`? 

Also, a few higher level thoughts about having `is_selective_syncer` on the resource's `service_config` itself (where it'd still give us a service-level `selectiveSyncer`):
1. We could get the `Resource` directly [here](https://github.com/viamrobotics/rdk/blob/c5bfc3ef574220cbea0d0e8c973908d4b908e8a8/services/datamanager/data_manager.go#L78) and then grab the sensor itself via `sensor, err := resource.(Sensor)`
2. As we're [having conversations](https://viaminc.slack.com/archives/C0307PGHR7E/p1698771409496899) about when `Reconfigure` is called and making it happen only when needed, right now the thought is to only have it reconfigure if the service is changed or one of its weak dependencies (via the `service_config`) does. If the underlying module is reconfigured but doesn't have a service_config, then that would be excluded.

Reply:
The test module has the model name specified [here](https://github.com/viam-labs/trigger-sync-examples-v2/blob/17d34b4ddd244ac337d3208a1c8b11208162d6ab/sync_module/selective_sync/selective_sync.go#L21) and then in the [config](https://github.com/viam-labs/trigger-sync-examples-v2/blob/37630c1774e016a3952295201f815cd981e5f094/sync_module/sync-on-color.json#L9).

Is the suggestion to have `is_selective_syncer` in the data capture config in lieu of having the equivalent in the datamanager config? If so, I'm not really convinced that's desirable. I feel like having this in datamanager config indicates at the top level the expected behavior of both sync and selective sync, whereas having this at the data capture indicates that this is somehow related datacapture and obfuscates the sync-level/ datamanger-level logic that is the selective sync "sensor" is actually operating on. Let me know if I misunderstood your question.

Reply:
Discussed offline, and my concern was with 1) grabbing the correct and resource (which can be answered by my other question about checking if it's possible to run RDK with multiple sensors of the same name, and if so, what happens here) and 2) reconfiguring the underlying component without having that trigger the Data Manager Service `Reconfigure`.

For the latter point, if the component changes in a way that no longer follows the expected scheme, that will fail at Sync() time which I think is expected behavior. As long as we update the selective syncer or enable/disable sync as expected, which would happen since the configuration at the DM Service level would change, then this works fine.